LanguageFeatures/Augmentation-libraries/augmented_expression_A02_t18_lib.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting setters: Within an augmenting setter `augmented` must be
+///   followed by an `=` and will directly invoke the augmented setter. If
+///   augmenting a field with a setter, this will invoke the implicit setter
+///   from the augmented field.
+///
+/// @description Checks that it is a compile-time error to declare local
+/// variable named `augmented` within an augmenting setter. Test an object
+/// pattern.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmented_expression_A02_t18.dart';
+
+augment void set topLevelSetter(String value) {
+    augmented? x;
```

Comment:
I'm afraid I still don't see any parameters here whose type has the name `augmented`. A local variable doesn't induce a setter, so we don't even have an implicitly induced setter whose parameter has that type.

Reply:
(Fixed in description)